node/src/components/contract_runtime/metrics.rs

```diff
@@ -0,0 +1,204 @@
+use prometheus::{self, Gauge, Histogram, IntGauge, Registry};
+
+use crate::{unregister_metric, utils};
+
+/// Value of upper bound of histogram.
+const EXPONENTIAL_BUCKET_START: f64 = 0.01;
+
+/// Multiplier of previous upper bound for next bound.
+const EXPONENTIAL_BUCKET_FACTOR: f64 = 2.0;
+
+/// Bucket count, with the last bucket going to +Inf which will not be included in the results.
+/// - start = 0.01, factor = 2.0, count = 10
+/// - start * factor ^ count = 0.01 * 2.0 ^ 10 = 10.24
+/// - Values above 10.24 (f64 seconds here) will not fall in a bucket that is kept.
+const EXPONENTIAL_BUCKET_COUNT: usize = 10;
+
+const RUN_EXECUTE_NAME: &str = "contract_runtime_run_execute";
+const RUN_EXECUTE_HELP: &str = "time in seconds to execute but not commit a contract";
+
+const APPLY_EFFECT_NAME: &str = "contract_runtime_apply_commit";
+const APPLY_EFFECT_HELP: &str = "time in seconds to commit the execution effects of a contract";
+
+const COMMIT_UPGRADE_NAME: &str = "contract_runtime_commit_upgrade";
+const COMMIT_UPGRADE_HELP: &str = "time in seconds to commit an upgrade";
+
+const RUN_QUERY_NAME: &str = "contract_runtime_run_query";
+const RUN_QUERY_HELP: &str = "time in seconds to run a query in global state";
+
+const COMMIT_STEP_NAME: &str = "contract_runtime_commit_step";
+const COMMIT_STEP_HELP: &str = "time in seconds to commit the step at era end";
+
+const GET_BALANCE_NAME: &str = "contract_runtime_get_balance";
+const GET_BALANCE_HELP: &str = "time in seconds to get the balance of a purse from global state";
+
+const GET_VALIDATOR_WEIGHTS_NAME: &str = "contract_runtime_get_validator_weights";
+const GET_VALIDATOR_WEIGHTS_HELP: &str =
+    "time in seconds to get validator weights from global state";
+
+const GET_ERA_VALIDATORS_NAME: &str = "contract_runtime_get_era_validators";
+const GET_ERA_VALIDATORS_HELP: &str =
+    "time in seconds to get validators for a given era from global state";
+
+const GET_BIDS_NAME: &str = "contract_runtime_get_bids";
+const GET_BIDS_HELP: &str = "time in seconds to get bids from global state";
+
+const MISSING_TRIE_KEYS_NAME: &str = "contract_runtime_missing_trie_keys";
+const MISSING_TRIE_KEYS_HELP: &str = "time in seconds to get missing trie keys";
+
+const PUT_TRIE_NAME: &str = "contract_runtime_put_trie";
+const PUT_TRIE_HELP: &str = "time in seconds to put a trie";
+
+const GET_TRIE_NAME: &str = "contract_runtime_get_trie";
+const GET_TRIE_HELP: &str = "time in seconds to get a trie";
+
+const CHAIN_HEIGHT_NAME: &str = "chain_height";
+const CHAIN_HEIGHT_HELP: &str = "current chain height";
+
+const EXEC_BLOCK_NAME: &str = "contract_runtime_execute_block";
+const EXEC_BLOCK_HELP: &str = "time in seconds to execute all deploys in a block";
+
+const LATEST_COMMIT_STEP_NAME: &str = "contract_runtime_latest_commit_step";
```

Comment:
I know it hasn't been the style so far but we should be including unit in the metric's name. In Grafana, there are no comments - i.e. the `LATEST_COMMIT_STEP_HELP` is not available.